src/service/controllers/voice.js

```diff
@@ -92,6 +93,32 @@ reportCallState = call => {
 
 reportCallStateForAgent = (userName, call) => {
   reporting.setSubject(userName, "Calls", { id: call.id, state: call.state });
+  changeAgentStatus(userName, call);
+};
+
+changeAgentStatus = (userName, call) => {
+	const user = conf.userByName(userName);
+	if (user && user.activeSession && user.activeSession.dn) {
+		const dn = user.activeSession.dn;
+		if (call.state === 'Established') {
```

Comment:
As we have several if else with call.state, it would be cleaner to use a switch case

Reply:
Refactoring to be done by a separate pull request, I guess